SEGFAULT: describe ingress with backend.resource

**What happened**:
SEGFAULT
```console
➜  root git:(xxxxx) ✗ kubectl -n base describe ing nginx-xxxxxx-t95n1d2d
Warning: extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5560a0d4e1dc]

goroutine 1 [running]:
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe.(*IngressDescriber).describeBackendV1(0xc000707df0, 0xc00068f530, 0x4, 0xc000e13960, 0xc000569900, 0x1)
        k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe/describe.go:2458 +0x7c
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe.(*IngressDescriber).describeIngressV1.func1(0x5560a153a400, 0xc00059a370, 0xc00056bcb0, 0x0)
        k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe/describe.go:2503 +0x885
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe.tabbedString(0xc000e13a60, 0x0, 0x0, 0x0, 0x0)
        k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe/describe.go:4890 +0xbe
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe.(*IngressDescriber).describeIngressV1(0xc000707df0, 0xc0005fd680, 0xc00091eaf0, 0xc0005fd680, 0xc00091eaf0, 0x0, 0x0)
        k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe/describe.go:2464 +0x68
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe.(*IngressDescriber).Describe(0xc000707df0, 0x7ffe00164061, 0x4, 0x7ffe00164073, 0x15, 0x1, 0x203000, 0x203000, 0x0, 0x5560a228fde0)
        k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe/describe.go:2394 +0x10f
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/describe.(*DescribeOptions).Run(0xc0007a74a0, 0x0, 0x5560a15150e0)
        k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/describe/describe.go:188 +0x4df
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/describe.NewCmdDescribe.func1(0xc0007aa840, 0xc00014f240, 0x2, 0x4)
        k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/describe/describe.go:110 +0xa5
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc0007aa840, 0xc00014f140, 0x4, 0x4, 0xc0007aa840, 0xc00014f140)
        k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:854 +0x2c2
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0000f2b00, 0xc00003c240, 0xc00003c1e0, 0x6)
        k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:958 +0x375
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:895
main.main()
        _output/local/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:49 +0x21f
```

Also, I'm not using `extensions/v1beta1 Ingress`, Should the warning be here ?

**What you expected to happen**:
describe the ingress, even if it doesn't make sense.


**How to reproduce it (as minimally and precisely as possible)**:
Create an ingress with `backend.resource` targeted at an AWS EFS instead of using `backend.service` that redirects to a basic nginx container.
Note: I tried to use `backend.resource` but I really don't know what I'm doing, the documentation is... sparse to say the least.
Note bis: I'm using pulumi (typescript) and `--export` has been deprecated, I already spent too much time creating this issue, not gonna translate to yaml.
```typescript
import * as k8s from "@pulumi/kubernetes";

  const ingress = new k8s.networking.v1.Ingress("my-ingress", {
    metadata: {namespace: "my-super-cool-namespace"},
    spec: {
      rules: [{
        host: `logs.acmecorp.com`,
        http: {
          paths: [{
            pathType: "Prefix",
            path: "/",
            backend: {
              resource: {
                kind: "PersistentVolumeClaim",
                name: persistantVolumeClaim.metadata.apply(m => m.name),
              }
            }
          }]
        }
      }]
    }
  })
```

**Environment**:

- EKS 1.19
- pulumi v2.22.0 (@pulumi/kubernetes: 2.8.2)
- `kubectl version`:
```console
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"archive", BuildDate:"2021-01-13T22:07:54Z", GoVersion:"go1.15.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.6-eks-49a6c0", GitCommit:"49a6c0bf091506e7bafcdb1b142351b69363355a", GitTreeState:"clean", BuildDate:"2020-12-23T22:10:21Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```

--- a/staging/src/k8s.io/kubectl/pkg/describe/describe.go
+++ b/staging/src/k8s.io/kubectl/pkg/describe/describe.go
@@ -2556,7 +2556,11 @@ func (i *IngressDescriber) describeBackendV1(ns string, backend *networkingv1.In
 	}
 	if backend.Resource != nil {
 		ic := backend.Resource
-		return fmt.Sprintf("APIGroup: %v, Kind: %v, Name: %v", *ic.APIGroup, ic.Kind, ic.Name)
+		apiGroup := "<none>"
+		if ic.APIGroup != nil {
+			apiGroup = fmt.Sprintf("%v", *ic.APIGroup)
+		}
+		return fmt.Sprintf("APIGroup: %v, Kind: %v, Name: %v", apiGroup, ic.Kind, ic.Name)
 	}
 	return ""
 }

--- a/staging/src/k8s.io/kubectl/pkg/describe/describe_test.go
+++ b/staging/src/k8s.io/kubectl/pkg/describe/describe_test.go
@@ -2213,6 +2213,12 @@ func TestDescribeIngress(t *testing.T) {
 			Name:     "bar",
 		},
 	}
+	backendResourceNoAPIGroup := networkingv1.IngressBackend{
+		Resource: &corev1.TypedLocalObjectReference{
+			Kind: "foo",
+			Name: "bar",
+		},
+	}
 
 	tests := map[string]struct {
 		input  *fake.Clientset
@@ -2279,6 +2285,42 @@ Rules:
   ----         ----  --------
   foo.bar.com  
                /foo   APIGroup: example.com, Kind: foo, Name: bar
+Annotations:   <none>
+Events:        <none>` + "\n",
+		},
+		"IngressRule.HTTP.Paths.Backend.Resource v1 Without APIGroup": {
+			input: fake.NewSimpleClientset(&networkingv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "bar",
+					Namespace: "foo",
+				},
+				Spec: networkingv1.IngressSpec{
+					Rules: []networkingv1.IngressRule{
+						{
+							Host: "foo.bar.com",
+							IngressRuleValue: networkingv1.IngressRuleValue{
+								HTTP: &networkingv1.HTTPIngressRuleValue{
+									Paths: []networkingv1.HTTPIngressPath{
+										{
+											Path:    "/foo",
+											Backend: backendResourceNoAPIGroup,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}),
+			output: `Name:             bar
+Namespace:        foo
+Address:          
+Default backend:  default-http-backend:80 (<error: endpoints "default-http-backend" not found>)
+Rules:
+  Host         Path  Backends
+  ----         ----  --------
+  foo.bar.com  
+               /foo   APIGroup: <none>, Kind: foo, Name: bar
 Annotations:   <none>
 Events:        <none>` + "\n",
 		},

